### PR TITLE
Firing pin acquisition 2: electric boogaloo

### PR DESCRIPTION
**Pull request by OneOneThreeEight**
I just wanted to make that title, not that it actually fit the situation.

* Adds a box of firing pins to the RD's locker. Can't permit mass-production of firing pins via R&D, sorry lads, this is the best anyone can do.
* This box of pins has a varied assortment of pin types.
* Pin removal is based on probability as the screwdriver is the only means of removing pins at the moment. When I figure out what would be a proper tool to do it reliably, and come up with unique sprites for it, ~~I'll change this PR. Right now it's just proof of concept.~~ it's out of scope atm since I will be making it an R&D-produced swiss army knife instead.

### CODE REVIEW

**Comment by OneOneThreeEight** [comment]
!wip

**Comment by OneOneThreeEight** [comment]
@Arrow768 I'll be adding a specific grasper tool that can be produced through R&D to precisely remove pins. It'll require a fair deal of research and about half a diamond to produce. I'll probably also add other neat features so it later down the line (once i figure out some applications, that is) so it isn't specialized to a single task.

**Comment by skull132** [comment]
In general, ye sure. Whenever it's done.

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -725,7 +725,7 @@ obj/item/weapon/gun/Destroy()
 	if(isscrewdriver(I))
 		visible_message("<span class = 'warning'>[user] begins to try and pry out [src]'s firing pin!</span>")
 		if(do_after(user,45 SECONDS,act_target = src))
-			if(pin.durable)
+			if(prob(50))
```
Comment (Arrow768) [inline code/modules/projectiles/gun.dm]:
```suggestion
			if(pin.durable || prob(50))
```
To respect the durable value so there is a 100% chance to remove durable pins

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 2]
```diff
@@ -385,6 +385,21 @@
...
 
+/obj/item/weapon/storage/box/firingpinsRD
+	name = "box of assorted firing pins"
+	desc = "A box of varied assortment of firing pins. Appears to have R&D stickers on all sides of the box. Also seems to have a smiley face sticker on the top of it."
+
+/obj/item/weapon/storage/box/firingpins/fill()
+	..()
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin(src)
+	new /obj/item/device/firing_pin/implant(src)
```
Comment (Arrow768) [inline code/game/objects/items/weapons/storage/boxes.dm]: The `/obj/item/device/firing_pin/implant` pins are useless ingame, as you wont be able to fire them.
Replace them with `/obj/item/device/firing_pin/access`

Comment (OneOneThreeEight) [reply code/game/objects/items/weapons/storage/boxes.dm]: Changes made.

**Pull request merged**